### PR TITLE
feat: use rollup-plugin-commonjs

### PR DESCRIPTION
**Pull request by DDeis**
As evoqued in #73 and #62

### CODE REVIEW

**Comment by DDeis** [comment]
Include `node_modules/rxjs/**` only and add the possibility to include other commonjs dependencies manually with `ng-package.json` to #62 problem.

**Comment by dherges** [comment]
Hi @DDeis,

I really like this improvement!

I suggest the following: we have integration testing samples (in the folder `integration` obvious). Add a test (a service or component or whatever) to the `src` that
 - a) uses the `import { map } from 'rxjs/operator/map';` syntax
 - b) uses the "old-fashioned"(?) `import 'rxjs/add/operator/map;` syntax

Then, the `angular.service.ts` used to demonstrate usage of the `lib.externals` property for the syntax in a. If possible, this test should be kept by another kind of import.

If you like, you can also create another sample, e.g. `integration/sample_rxjs` that demonstrates the issues you had (the motivation for the change) and the two syntax-ways of using rxjs operators (see above: a, b)

**Comment by DDeis** [comment]
Hi @dherges, thank you for your feedbacks, I can't work on this weekend but I will update the PR follow ONG your recommandations as soon as I can and will check if 'rxjs/add/operator/*' Stills works.

Also I wanted to know what was your opinion about the knew 'commonjs' property in 'ng-package', do you think it's a proper name ? Do you think we we can include this property in 'externals' and create a new property inside for "old fashionned" symbols to have both ways to include together ?

**Comment by DDeis** [comment]
Added sample to demonstrate the two syntax-ways of using rxjs operators as suggested.

**Comment by dherges** [comment]
@DDeis Hi, thanks for your effort! I will look at it.

**Comment by dherges** [comment]
@DDeis I made some smaller changes (tab indentation, update the description in the json schema).

However, when trying out this version in a larger project, the build errors:


```bash
BUILD ERROR
'BehaviorSubject' is not exported by ..\..\node_modules\rxjs\BehaviorSubject.js
Error: 'BehaviorSubject' is not exported by ..\..\node_modules\rxjs\BehaviorSubject.js
    at error (....\node_modules\ng-packagr\node_modules\rollup\dist\rollup.js:177:12)
    at Module.error$1 [as error] (....\node_modules\ng-packagr\node_modules\rollup\dist\rollup.js:8434:2)
    at Module.trace (....\node_modules\ng-packagr\node_modules\rollup\dist\rollup.js:8536:9)
    at ModuleScope.findDeclaration (....\node_modules\ng-packagr\node_modules\rollup\dist\rollup.js:8114:22)
    at Scope.findDeclaration (....\node_modules\ng-packagr\node_modules\rollup\dist\rollup.js:5907:39)
    at Node.bind (....\node_modules\ng-packagr\node_modules\rollup\dist\rollup.js:7165:29)
    at ....\node_modules\ng-packagr\node_modules\rollup\dist\rollup.js:5701:50
    at Node.eachChild (....\node_modules\ng-packagr\node_modules\rollup\dist\rollup.js:5718:5)
    at Node.bind (....\node_modules\ng-packagr\node_modules\rollup\dist\rollup.js:5701:7)
    at ....\node_modules\ng-packagr\node_modules\rollup\dist\rollup.js:5701:50

error Command failed with exit code 111.
```

File: lib/steps/rollup.ts [hunk 1]
```diff
@@ -27,151 +29,26 @@ export const rollup = (opts: RollupOptions) => {
...
+    // external symbols passed from the user's ng-package.json
     ...opts.externals,
   };
 
+  const ROLLUP_COMMONJS_INCLUDE = [
+    // RxJS dependencies
+    'node_modules/rxjs/**',
+    // commonjs symbols passed from the user's ng-package.json
+    ...opts.commonjs
+  ];
```
Comment (dherges) [inline lib/steps/rollup.ts]: `rollup-plugin-commonjs` does have an issue w/ monorepos.

```bash
$ cd packages/my-package
$ ../../node_modules/.bin/ng-packagr
```

Now, rollup will resolve the dependency to `../../node_modules/rxjs` but the value in `ROLLUP_COMMONJS_INCLUDE ` does not match.

In result, together w/ removing the default `externals` breaks existing builds.

**Comment by DDeis** [comment]
Update the PR to allow specifying the path of the `node_modules` directory.
Would this solve the issue with monorepos ?

**Comment by dherges** [comment]
That still breaks existing builds w/ their current configuration. So the change is requiring people to update their config in a project where they were building rxjs dependencies just fine!

Altogether, there are two things:
 - rxjs and their two import styles #73 
 - commonjs support #62 

Is it possible to separate the changes for these two? One for the rxjs thing, one for the commonjs.

**Comment by DDeis** [comment]
Right, maybe we should have two PRs to address these issues distinctively:
- rxjs and their two import styles #73: adding missing rxjs dependencies
- commonjs support #62: using rollup-plugin-commonjs

**Comment by DDeis** [comment]
PR splited into: 
- PR #82: rxjs two import styles #73
- PR #83: commonjs support #62

**Pull request closed without merging**